src/odemis/driver/orsay.py

```diff
@@ -0,0 +1,880 @@
+                             "datamodel.HybridPlatform.AnalysisChamber.Pressure. Parameter passed is %s"
+                             % parameter.Name)
+        if attributeName != "Actual":
+            return
+        self.pressure._set_value(float(parameter.Actual), force_write=True)
+
+    def _changeVacuum(self, goal, wait=True):
+        """
+        goal (int): goal state of the vacuum: (0: "vented", 1: "primary vacuum", 2: "high vacuum")
+        wait (bool): if True, the function will block until the goal vacuum state is reached
+        return (int): actual state of the vacuum at the end of this function: (0: "vented", 1: "primary vacuum",
+                      2: "high vacuum")
+
+        Sets the vacuum status on the Orsay server to argument goal and waits until it is reached.
+        Then returns the reached vacuum status.
+        """
+        logging.debug("Setting vacuum status to %s." % self.axes["vacuum"].choices[goal])
+        self._chamber.VacuumStatus.Target = goal
+        if wait:
+            self._vacuumStatusReached.wait()
+        return self._chamber.VacuumStatus.Actual
+
+    def _changeGateOpen(self, goal):
+        """
+        goal (bool): goal position of the gate: (True: "open", False: "closed")
+        return (bool): goal position of the gate as set to the server: (True: "open", False: "closed")
+
+        Opens ValveP5 on the Orsay server if argument goal is True. Closes it otherwise.
+        """
+        logging.debug("Setting gate to %s." % ("open" if goal else "closed"))
+        self._gate.IsOpen.Target = VALVE_OPEN if goal else VALVE_CLOSED
+        return self._gate.IsOpen.Target == VALVE_OPEN
+
+    @isasync
+    def moveAbs(self, pos, wait=True):
+        """
+        Move the axis of this actuator to pos.
+        """
+        self._checkMoveAbs(pos)
+        return self._executor.submit(self._changeVacuum, goal=pos["vacuum"], wait=wait)
+
+    @isasync
+    def moveRel(self, shift):
+        """
+        Move the axis of this actuator by shift.
+        """
+        pass
+
+    def stop(self, axes=None):
+        """
+        Stop changing the vacuum status
+        """
+        if not axes or "vacuum" in axes:
+            logging.debug("Stopping vacuum.")
+            self.parent.datamodel.HybridPlatform.Stop.Target = COMPONENT_STOP
+            self.parent.datamodel.HybridPlatform.Cancel.Target = True
+            self._executor.cancel()
+
+    def terminate(self):
+        """
+        Called when Odemis is closed
+        """
+        self._gate.ErrorState.Unsubscribe(self._updateErrorState)
+        self._chamber.VacuumStatus.Unsubscribe(self._updatePosition)
+        self._chamber.Pressure.Unsubscribe(self._updatePressure)
+        self._gate.IsOpen.Unsubscribe(self._updateGateOpen)
+        if self._executor:
+            self._executor.shutdown()
+            self._executor = None
+        _gate = None
+        _chamber = None
```

Comment:
`self.` missing?
How come it's not picked up in your test cases? Do you call `.terminate()`?